src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -805,4 +809,56 @@ export class PatientViewPageStore {
         this.ajaxErrors = [];
     }
 
+    readonly trialMatches = remoteData<ITrialMatch[]>({
+        invoke: () => {
+            return postMatchMinerTrialMatches({mrn: this.patientId});
+        },
+        onError: (err: Error) => {
+            // fail silently
+        }
+    }, []);
+
+    readonly trialIds = remoteData<ITrialQuery>({
+        await: () => [
+            this.trialMatches
+        ],
+        invoke: async() => {
+            let nctIds = new Set<string>();
+            let protocolNos = new Set<string>();
+            _.forEach(this.trialMatches.result, (trialMatch: ITrialMatch) => {
+                if (_.isEmpty(trialMatch.protocolNo)) {
+                    nctIds.add(trialMatch.nctId);
+                } else {
+                    protocolNos.add(trialMatch.protocolNo);
+                }
+            });
+            return {
+                nct_id: [...nctIds],
+                protocol_no: [...protocolNos]
+            };
+        },
+        onError: (err: Error) => {
+            // fail silently
+        }
+    }, {
+        nct_id: [],
+        protocol_no: []
```

Comment:
does "protocol_no" mean protocolor_number?   if so, lets spell it out: protocol_number or protocol_id

Reply:
Sorry, I cannot change the protocol_no in here. This object will be sent to api for querying trials. Besides, protocol_no is defined in Clinical Trial Markup Language so I use the same name in frontend.

Reply:
ok